### PR TITLE
Add Testing for YDS Scores #11

### PR DESCRIPTION
**Pull request by DarrenZLew**
Fix for https://github.com/OpenBeta/sandbag/issues/11

- Improved the regex logic for the YDS grade scale to be more specific to only supporting. Used a slightly modified version of the regex found here for our use cases (https://openbeta.substack.com/p/climbing-grades-in-javascript). Thanks Viet :)
1. 5.0 to 5.16 with + and -
2. 5.10x to 5.16x where x can be a, b, c, or d. Allows for slash grade such as 5.10a/b

- Updated tests to yds.ts for un-normalized YDS scores. Displayed console warnings if the provided grade was not supported for YDS.
- Added win-node-env package for windows developers to get the tests working in a windows OS.

### CODE REVIEW

**Comment by vnugent** [comment]
Thanks for this.  I'm not familiar with `win-node-env`.  It seems to cause an error in the GH test action (which runs on Ubuntu).

https://github.com/OpenBeta/sandbag/runs/8221521771?check_suite_focus=true

**Comment by DarrenZLew** [comment]
> Thanks for this. I'm not familiar with `win-node-env`. It seems to cause an error in the GH test action (which runs on Ubuntu).
> 
> https://github.com/OpenBeta/sandbag/runs/8221521771?check_suite_focus=true

Apparently `win-node-env` will not work with other OS's, according to https://www.npmjs.com/package/win-node-env. It says we could add it in the `optionalDependencies` to get around this for CI/CD. I pushed a new commit updating it to be an optionalDependency. Could you see if this still causes the build to fail?

**Comment by vnugent** [comment]
Is `win-node-env` needed because of [this line](https://github.com/OpenBeta/sandbag/blob/develop/package.json#L12)?

would https://www.npmjs.com/package/cross-env be a better alternative?

**Comment by DarrenZLew** [comment]
> Is `win-node-env` needed because of [this line](https://github.com/OpenBeta/sandbag/blob/develop/package.json#L12)?
> 
> would https://www.npmjs.com/package/cross-env be a better alternative?

Yes, it was due to that line. I tried using `cross-env` before, but it didn't seem to work. Let me try again though.

Sorry for this difficulty. I normally use linux, so any issues for Windows OS is new to me.

**Comment by vnugent** [comment]
Tests are running fine now.  I'm ok to merge this if you are.  We can revisit this env issue later.

**Comment by vnugent** [comment]
@all-contributors add @DarrenZLew for code and ideas
